### PR TITLE
RESTDataSource docs updates

### PR DESCRIPTION
**Pull request by trevor-scheer**
Updates for correctness based on upcoming v5 changes

### CODE REVIEW

File: docs/source/data/fetching-rest.mdx [hunk 1]
```diff
@@ -63,7 +63,7 @@ class MoviesAPI extends RESTDataSource {
 
 </MultiCodeBlock>
 
-You can extend the `RESTDataSource` class to implement whatever data-fetching methods your resolvers need. These methods should use the built-in convenience methods (e.g., `get` and `post`) to perform HTTP requests, helping you add query parameters, parse and cache JSON results, dedupe requests, and handle errors.
+You can extend the `RESTDataSource` class to implement whatever data-fetching methods your resolvers need. These methods should use the built-in convenience methods (e.g., `get` and `post`) to perform HTTP requests, helping you add query parameters, parse and cache JSON results, dedupe requests, and handle errors. More complex use cases can use the `fetch` method directly. The `fetch` method returns both the parsed body as well as the response object which affords more flexibility for use cases like reading response headers.
```
Comment (StephenBarlow) [inline docs/source/data/fetching-rest.mdx:66]:
```suggestion
You can extend the `RESTDataSource` class to implement whatever data-fetching methods your resolvers need. These methods should use the built-in convenience methods (e.g., `get` and `post`) to perform HTTP requests, helping you add query parameters, parse and cache JSON results, dedupe requests, and handle errors. More complex use cases can use the `fetch` method directly. The `fetch` method returns both the parsed body and the response object, which provides more flexibility for use cases like reading response headers.
```

File: docs/source/data/fetching-rest.mdx [hunk 2]
```diff
@@ -136,18 +136,21 @@ const resolvers = {
 
 The `RESTDataSource` class provides its subclasses with _two_ layers of caching:
 
-- The first layer **stores all outgoing `GET` requests** by their URLs in a separate memoized cache.
+- The first layer deduplicates _concurrent_ outgoing `GET` (and `HEAD`) requests by default. Deduplication is keyed on the request's method and URL. This behavior can be configured by overriding the `requestDeduplicationPolicyFor` method. For more details on how to configure this please consult [the README](https://github.com/apollographql/datasource-rest#requestdeduplicationpolicyfor).
```
Comment (StephenBarlow) [inline docs/source/data/fetching-rest.mdx:139]:
```suggestion
- The first layer deduplicates _concurrent_ outgoing `GET` (and `HEAD`) requests by default. Deduplication is keyed on the request's method and URL. You can configure this behavior by overriding the `requestDeduplicationPolicyFor` method. For more details, see [the README](https://github.com/apollographql/datasource-rest#requestdeduplicationpolicyfor).
```

File: docs/source/data/fetching-rest.mdx [hunk 3]
```diff
@@ -136,18 +136,21 @@ const resolvers = {
...
+
+> Note: in versions prior to v5 of `RESTDataSource`, _all_ outgoing `GET` requests were deduplicated. This behavior can be restored using the `deduplicate-until-invalidated` policy (explained more in-depth in [the README](https://github.com/apollographql/datasource-rest#requestdeduplicationpolicyfor)).
```
Comment (StephenBarlow) [inline docs/source/data/fetching-rest.mdx:141]:
```suggestion
> **Note:** In versions of `RESTDataSource` prior to v5, _all_ outgoing `GET` requests are deduplicated. You can achieve this same behavior with the `deduplicate-until-invalidated` policy (explained further in [the README](https://github.com/apollographql/datasource-rest#requestdeduplicationpolicyfor)).
```

File: docs/source/data/fetching-rest.mdx [hunk 4]
```diff
@@ -136,18 +136,21 @@ const resolvers = {
...
 
-### `GET` requests and responses
+### `GET` (and `HEAD`) requests and responses
 
-Every time you create a new instance of a `RESTDataSource` subclass, under the hood, it makes an internal memoized cache. By default, `RESTDataSource` [automatically stores](https://github.com/apollographql/datasource-rest/blob/e9c6e1bd5498a44fce3046aeb6556ef9d4c0fc6a/src/RESTDataSource.ts#L263) _all_ outgoing `GET` requests (by their URLs) alongside their results in this internal cache.
+Every time you create a new instance of a `RESTDataSource` subclass, under the hood, it makes an internal cache. By default, `RESTDataSource` automatically deduplicates _concurrent_ `GET` (and `HEAD`) requests (keyed by their method and URLs) alongside their results in this internal cache. This is referred to as request deduplication. The default behavior can be configured by overriding the [`requestDeduplicationPolicyFor`](https://github.com/apollographql/datasource-rest/blob/main/README.md#requestdeduplicationpolicyfor) method on the class.
```
Comment (StephenBarlow) [inline docs/source/data/fetching-rest.mdx:149]:
```suggestion
Every time you instantiate a `RESTDataSource` subclass, under the hood that instance creates an internal cache. By default, `RESTDataSource` automatically deduplicates _concurrent_ `GET` (and `HEAD`) requests (keyed by their method and URLs) alongside their results in this internal cache. This behavior is called request deduplication. You can configure this default behavior by overriding the [`requestDeduplicationPolicyFor`](https://github.com/apollographql/datasource-rest/blob/main/README.md#requestdeduplicationpolicyfor) method on the class.
```

File: docs/source/data/fetching-rest.mdx [hunk 5]
```diff
@@ -172,9 +175,9 @@ If any resolver in the current operation attempts a parallel `GET` request to th
 
 > This internal caching mechanism is why we create a new `RESTDataSource` instance for _every_ request. Otherwise, responses would be cached across requests even if they specify they shouldn't be!
 
-You can change how `GET` requests are stored in `RESTDataSource`'s memoized cache by overwriting the [`cacheKeyFor` method](https://github.com/apollographql/datasource-rest#cachekeyfor).
+You can change how `GET` (and `HEAD`) requests are stored in `RESTDataSource`'s deduplication cache by overwriting the [`cacheKeyFor` method](https://github.com/apollographql/datasource-rest#cachekeyfor). By default, the cache key is the method and URL of the request.
```
Comment (StephenBarlow) [inline docs/source/data/fetching-rest.mdx:178]:
```suggestion
You can change how `GET` (and `HEAD`) requests are stored in `RESTDataSource`'s deduplication cache by overwriting the [`cacheKeyFor` method](https://github.com/apollographql/datasource-rest#cachekeyfor). By default, a request's cache key is the combination of its HTTP method and URL.
```

File: docs/source/data/fetching-rest.mdx [hunk 6]
```diff
@@ -172,9 +175,9 @@ If any resolver in the current operation attempts a parallel `GET` request to th
...
 
-If you'd like to disable `GET` request caching you can set `memoizeGetRequests` to `false`:
+If you'd like to restore the deduplication policy from before `RESTDataSource` v5, you can configure `requestDeduplicationPolicyFor` like so:
```
Comment (StephenBarlow) [inline docs/source/data/fetching-rest.mdx:180]:
```suggestion
To restore the deduplication policy from before `RESTDataSource` v5, you can configure `requestDeduplicationPolicyFor` like so:
```

**Review by StephenBarlow — APPROVED** [review]
Approved with some edits!

File: docs/source/data/fetching-rest.mdx [hunk 7]
```diff
@@ -7,7 +7,7 @@ import TopLevelAwait from "../shared/top-level-await.mdx"
 
 > See the [`@apollo/datasource-rest` page](https://github.com/apollographql/datasource-rest) for the full details of the `RESTDataSource` API.
```
Comment (glasser) [inline docs/source/data/fetching-rest.mdx:8]:
```suggestion
> See the [`@apollo/datasource-rest` README](https://github.com/apollographql/datasource-rest) for the full details of the `RESTDataSource` API.
```
perhaps?

**Review by glasser — APPROVED** [review]

**Pull request merged**